service/worker/scheduler/workflow.go

```diff
@@ -334,6 +356,28 @@ func (s *scheduler) processPatch(patch *schedpb.SchedulePatch) {
 	}
 }
 
+func (s *scheduler) getNextTime(after time.Time) getNextTimeResult {
+
+	// we populate the map sequentially, if after is not in the map, it means we either exhausted
+	// all items, or we jumped through time (forward or backward), in either case, refresh the cache
+	next, ok := s.nextTimeResultCache[after]
+	if ok {
+		return next
+	}
+	// Run this logic in a SideEffect so that we can fix bugs there without breaking
+	// existing schedule workflows.
+	panicIfErr(workflow.SideEffect(s.ctx, func(ctx workflow.Context) interface{} {
+		results := make(map[time.Time]getNextTimeResult)
+		for t := after; !t.IsZero() && len(results) < maxNextTimeResultCacheSize; {
+			next := s.cspec.getNextTime(t)
+			results[t] = next
+			t = next.Next
+		}
+		return results
+	}).Get(&s.nextTimeResultCache))
```

Comment:
one more thing: I think that Get into a map may actually add entries instead of resetting the map (it ends up in the data converter which does json.Unmarshal), so the map might keep growing. I think we should explicitly clear the map before Get just in case